File: personal_project_back-end/common-service/src/main/java/com/example/common/service/models/requests/PagingQueryConditionRequest.java
package com.example.common.service.models.requests;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PagingQueryConditionRequest {
    private int pageSize = 10;
    private int pageCurrent = 0;
    private String sortBy;
    private String sortType;
    private String keyword;

    public PagingQueryConditionRequest(int pageSize, int pageCurrent) {
        this.pageSize = pageSize;
        this.pageCurrent = pageCurrent;
    }
}

File: personal_project_front-end/src/models/request/PagingQueryConditionRequest.ts
export type PagingQueryConditionRequest = {
  pageSize: number;
  pageCurrent: number;
  sortBy?: string;
  sortType?: string;
  keyword?: string;
};

export const defaultPagingQueryConditionRequest: PagingQueryConditionRequest = {
  pageSize: 1000,
  pageCurrent: 0,
};
